src/main/java/se/simonsoft/cms/indexing/properties/HandlerSubjectScheme.java

```diff
@@ -0,0 +1,151 @@
+        this.sourceReader = (XmlSourceReaderS9api) sourceReader;
+        XsltCompiler compiler = processor.newXsltCompiler();
+        XsltExecutable xsltCompiled;
+        try {
+            logger.info("Compiling subject scheme properties indexing XSL.");
+            xsltCompiled = compiler.compile(stylesheet);
+            logger.info("Compiled subject scheme properties indexing XSL.");
+        } catch (SaxonApiException e) {
+            throw new RuntimeException("Failed to compile subject scheme properties indexing XSL: " + e.getMessage(), e);
+        }
+        transformer = xsltCompiled.load();
+    }
+
+    @Inject
+    public void setItemContentBufferStrategy(ItemContentBufferStrategy contentStrategy) {
+        this.contentStrategy = contentStrategy;
+    }
+
+    @Override
+    public void handle(IndexingItemProgress progress) {
+        String excludePattern = "(cmsconfig.*:.*|svn:.*|abx:.*)";
+        CmsItemProperties itemProperties = progress.getProperties();
+        CmsItemId itemId = getItemId(progress);
+        if (itemId == null) return;
+        CmsItemPath itemRelPath = itemId.getRelPath();
+        if (itemRelPath == null) return;
+        CmsItemPath itemParentPath = itemRelPath.getParent();
+        if (itemParentPath == null) return;
+        XdmMap properties = new XdmMap();
+        for (String key : itemProperties.getKeySet()) {
+            String value = itemProperties.getString(key);
+            if (!Pattern.compile(excludePattern).matcher(key).matches()) {
+                properties = properties.put(new XdmAtomicValue(key), new XdmAtomicValue(value));
+            }
+        }
+        if (properties.isEmpty()) return;
+        indexItemProperties(progress, properties);
+    }
+
+    @Override
+    protected CmsItemId getItemId(IndexingItemProgress progress) {
+        CmsChangesetItem item = progress.getItem();
+        // Ignore non-file items as later on contentStrategy.getBuffer(...) since its entry is missing a "size" attribute.
+        if (!item.isFile()) return null;
+        CmsRepository repo = progress.getRepository();
+        CmsItemPath path = item.getPath();
+        return repo.getItemId().withRelPath(path);
+    }
+
+    @Override
+    public Set<Class<? extends IndexingItemHandler>> getDependencies() {
+        return Set.of();
+    }
+
+    private void indexItemProperties(IndexingItemProgress progress, XdmMap properties) {
+        XmlSourceDocumentS9api document = null;
+        RepoRevision revision = progress.getRevision();
+        String projectFolder = progress.getItem().getPath().getPathSegments().get(0);
+        Destination destination = new SAXDestination(new ContentHandlerToIndexFields(progress.getFields()));
+        LoggingErrorListener errorListener = new LoggingErrorListener();
+        SaxonMessageListener messageListener = new SaxonMessageListener();
+        try {
```

Comment:
I would prefer to move the logic that gets the subject scheme to a separate method and send buffer as parameter to indexItemProperties.

Reply:
Done.